File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace MyCompiler {
    class Program {
        public static void Main(string[] args) {

            // 배열 초기화
            int[] numbers = new int[9];

            // 배열 값 입력
            for(int i = 0; i < 9; i++)
            {
                int num = int.Parse(Console.ReadLine());

                numbers[i] = num;
            }

            // 오름차순으로 배열 정렬
            numbers = numbers.OrderBy(value => value).ToArray();

            // 총합과 타겟 합 구하기
            int totalSum = numbers.Sum();
            int targetSum = totalSum - 100;

            // pair 값 초기화
            (int first, int second) pair = (0, 0);

            // 타겟 합이 나오는 인덱스 값 찾고 pair에 저장하기
            for(int i = 0; i < 9; i++)
            {
                for(int j = i; j < 9; j++)
                {
                    if(numbers[i] + numbers[j] == targetSum)
                    {
                        pair.first = i; 
                        pair.second = j;
                        break;
                    }
                } 

                if(pair.first != 0) break;
            }

            // 일곱 난쟁이 찾기
            for(int i = 0; i < 9; i++)
            {
                if(i == pair.first || i == pair.second) continue;

                Console.WriteLine(numbers[i]);
            }
        }
    }
}

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cc
#include <bits/stdc++.h>

using namespace std;

int main() {

    // 값 저장할 배열
    vector<int> numbers;

    // 입력한 값 배열에 추가
    for(int i = 0; i < 9; i++)
    {
        int num = 0;
        cin >> num;
        numbers.push_back(num);
    }

    // 배열 값 최소값으로 정렬하기
    sort(numbers.begin(), numbers.end());


    // 모든 숫자 합한 후, 100을 빼서 여분 합 구하기
    int totalSum = accumulate(numbers.begin(), numbers.end(), 0);
    int targetSum = totalSum - 100;


    // 더해서 여분합이 나오는 인덱스 쌍 찾기
    pair<int ,int> exclude = {0, 0};

    for(int i = 0; i < 9; i++)
    {
        for(int j = i; j < 9; j++)
        {
            if(numbers[i] + numbers[j] == targetSum)
            {
                exclude = {i, j};
                break;
            }
        }
        if (exclude.first != 0) break;
    }


    // 일곱 난쟁이 찾기
    for(int i = 0; i < 9; i++)
    {
        if(i == exclude.first || i == exclude.second) continue;
        
        cout << numbers[i] << '\n';
    }
}